### PR TITLE
opencpn: 5.6.2 -> 5.8.4

### PR DESCRIPTION
**Pull request by philiptaron**
## Description of changes

`opencpn` is currently failing: https://hydra.nixos.org/build/246770144. This update fixes the build.

https://github.com/opencpn/opencpn/compare/Release_5.6.2...Release_5.8.4

All added packages are from looking through the `configurePhase` output.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -34372,9 +34372,9 @@ with pkgs;
 
   openbrf = libsForQt5.callPackage ../applications/misc/openbrf { };
 
-  opencpn = darwin.apple_sdk_11_0.callPackage ../applications/misc/opencpn {
```
Comment (drupol) [inline pkgs/top-level/all-packages.nix:34375]: Are you sure about this?

Comment (drupol) [reply pkgs/top-level/all-packages.nix:34375]: Fair enough then.

Comment (philiptaron) [reply pkgs/top-level/all-packages.nix:34375]: > However, I see in the logs that it's not possible to build it on Darwin, have you seen?

[x86_64-darwin](https://logs.ofborg.org/?key=nixos/nixpkgs.281450&attempt_id=b45c4d58-ac8c-451c-84a7-c4a7bce542fb):

```
CMake Error at /nix/store/101zq7jbr3w3q0v2f1xyz9y6d488f4lv-cmake-3.27.8/share/cmake-3.27/Modules/GoogleTestAddTests.cmake:112 (message):
  Error running test executable.
    Path: '/tmp/nix-build-opencpn-5.8.4.drv-0/source/build/test/tests'
    Result: Process terminated due to timeout
    Output:
      
Call Stack (most recent call first):
  /nix/store/101zq7jbr3w3q0v2f1xyz9y6d488f4lv-cmake-3.27.8/share/cmake-3.27/Modules/GoogleTestAddTests.cmake:225 (gtest_discover_tests_impl)
```

[aarch64-darwin](https://logs.ofborg.org/?key=nixos/nixpkgs.281450&attempt_id=d86dc85a-1af8-411b-b88f-fe901e3c840f)

```
CMake Error at /nix/store/slgfvfhi4nbdms9h7p13rp998ls191az-cmake-3.27.8/share/cmake-3.27/Modules/GoogleTestAddTests.cmake:112 (message):
  Error running test executable.
    Path: '/tmp/nix-build-opencpn-5.8.4.drv-0/source/build/test/tests'
    Result: Process terminated due to timeout
    Output:
      
Call Stack (most recent call first):
  /nix/store/slgfvfhi4nbdms9h7p13rp998ls191az-cmake-3.27.8/share/cmake-3.27/Modules/GoogleTestAddTests.cmake:225 (gtest_discover_tests_impl)
```

I ran these both successfully... so I don't know what's up.

Comment (drupol) [reply pkgs/top-level/all-packages.nix:34375]: @NixOS/darwin-maintainers Can you help in here?

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 281450` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>opencpn</li>
  </ul>
</details>

**Comment by wegank** [comment]
@ofborg build opencpn opencpn.passthru.tests

**Comment by philiptaron** [comment]
@wegank, it looks like it worked: https://logs.ofborg.org/?key=nixos/nixpkgs.281450&attempt_id=450c3659-add1-494e-a34e-3a3f4334a0f6

**Comment by philiptaron** [comment]
Ping on this: I don't think I have more work to do, and [the build is currently broken](https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.opencpn.x86_64-linux).

**Comment by philiptaron** [comment]
Thanks @wegank!

**Pull request merged**